### PR TITLE
Build with z5 and hdf5 on windows

### PR DESCRIPTION
**Pull request by constantinpape**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by constantinpape** [comment]
@conda-forge-admin, please rerender

**Comment by constantinpape** [comment]
The build fails with some compiler errors (Illegal Indirection), although I can build it in a [different CI](https://github.com/constantinpape/nifty/actions/runs/621822497) (although the CI later fails at the test stage).
 
 The difference is the compiler version, the conda-forge build uses MSVC 19.16, the other CI 19.28.
 
 Note that the conda-forge build seems to use the system compiler and not the compiler provided by conda:
 ```
 -- The C compiler identification is MSVC 19.16.27045.0
-- The CXX compiler identification is MSVC 19.16.27045.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Enterprise/VC/Tools/MSVC/14.16.27023/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Enterprise/VC/Tools/MSVC/14.16.27023/bin/Hostx64/x64/cl.exe - skipped
```
[see also here](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=285786&view=logs&j=171a126d-c574-5c8c-1269-ff3b989e923d&t=1183ba29-a0b5-5324-8463-2a49ace9e213&l=432)

So the first thing to try to fix this issue would be using the conda compiler (which would be more clean in any case).
Unfortunately I don't know how to do this on Windows...

**Comment by constantinpape** [comment]
@conda-forge-admin, please rerender

**Comment by constantinpape** [comment]
It turns out conda picks up the correct compiler, it's just too old.
To fix this, I tried to switch to `vs2019`, following the example [here](https://github.com/conda-forge/libignition-physics-feedstock/blob/c586d765a2f5fd0ecf6da43c53315c898c9bf6bd/recipe/conda_build_config.yaml#L6-L9).

This seems to pick the correct compiler, however the cmake call then fails because it does not find it any more:

```
(base) %SRC_DIR%>set "cxx_compiler=vs2019"

...

(%BUILD_PREFIX%) %SRC_DIR%\build>cmake .. -G "NMake Makefiles"          -DCMAKE_BUILD_TYPE=Release          -DCMAKE_PREFIX_PATH="%PREFIX%\Library"          -DCMAKE_INSTALL_PREFIX="%PREFIX%\Library"          -DBOOST_ROOT="%PREFIX%\Library"          -DBUILD_NIFTY_PYTHON=ON          -DWITH_HDF5=ON          -DWITH_Z5=ON          -DWITH_ZLIB=ON          -DWITH_BLOSC=ON          -DPYTHON_EXECUTABLE="%PREFIX%\python.exe" 
CMake Error at D:/bld/nifty_1619731407973/_build_env/Library/share/cmake-3.20/Modules/CMakeDetermineCCompiler.cmake:49 (message):
  Could not find compiler set in environment variable CC:

  cl.exe.
Call Stack (most recent call first):
  CMakeLists.txt:4 (project)
-- Configuring incomplete, errors occurred!
See also "D:/bld/nifty_1619731407973/work/build/CMakeFiles/CMakeOutput.log".


CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
```

**Comment by constantinpape** [comment]
@conda-forge-admin, please rerender

File: .azure-pipelines/azure-pipelines-win.yml [hunk 1]
```diff
@@ -8,14 +8,14 @@ jobs:
     vmImage: vs2017-win2016
```
Comment (wolfv) [inline .azure-pipelines/azure-pipelines-win.yml]: this shows that you're not re-rendering properly or something else is wrong.

This should say what you have configured in the conda-forge yaml config file.

File: recipe/conda-forge.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+azure:
```
Comment (wolfv) [inline recipe/conda-forge.yml]: this needs to sit at the root of the feedstock

Comment (wolfv) [reply recipe/conda-forge.yml]: pls check https://github.com/conda-forge/libignition-physics-feedstock/commit/c586d765a2f5fd0ecf6da43c53315c898c9bf6bd#diff-478786365dd93f740eb520c2faa03d7a0623273663b75472272c8ef94297bbe2

Comment (constantinpape) [reply recipe/conda-forge.yml]: Thanks! I moved it to root.

**Comment by constantinpape** [comment]
@conda-forge-admin, please rerender

**Pull request merged**